File: core/FlattenBinaryTreeToLinkedList/FlattenBinaryTreeToLinkedList.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def flatten(root: TreeNode) -> None:
    if not root:
        return
    stack, prev = [root], None
    while stack:
        cur = stack.pop()
        if prev:
            prev.right = cur
            prev.left = None
        if cur.right:
            stack.append(cur.right)
        if cur.left:
            stack.append(cur.left)
        prev = cur

File: core/FlattenBinaryTreeToLinkedList/FlattenBinaryTreeToLinkedList.java
package com.dbc.code;

import java.util.Stack;

public class FlattenBinaryTreeToLinkedList {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public void flatten(TreeNode root) {
        if (root == null) return;
        Stack<TreeNode> stack = new Stack<>();
        stack.add(root);
        TreeNode prev = null;
        while (!stack.isEmpty()) {
            TreeNode node = stack.pop();
            if (prev != null) {
                prev.right = node;
                prev.left = null;
            }
            if (node.right != null) stack.add(node.right);
            if (node.left != null) stack.add(node.left);
            prev = node;
        }
    }
}
